### DISCUSSION TITLE
[BUG] [PRISMA STUDIO] [PRISMA GENERATE]

### DISCUSSION THREAD

**Discussion by nathanmota-dev** [prisma/prisma#23806] [Q&A]
### Bug description

Message: Error in Prisma Client request: 

```
Invalid `f=e.match(Jyt)?.[1]??"",g=e.match(Qyt)?.[1]??null,v=e.match(Zyt)?.[1]??null,{getPrismaClient:E,PrismaClientKnownRequestError:b,PrismaClientRustPanicError:S,PrismaClientInitializationError:D,PrismaClientValidationError:C}=require(`${u.prismaClient}/runtime/${c}`),O=e,I=(0,f$.createHash)("sha256").update()` invocation in
C:\Users\natha\Desktop\git\special-topics-in-database\server\node_modules\prisma\build\index.js:1914:10334

  1911       }
  1912     }
  1913   }
→ 1914 `}});return Le.resourceList(a.workspaces)}};var _be=require("@prisma/engines");var t8e=require("buffer");function r8e(e,r,n,i){Object.defineProperty(e,r,{get:n,set:i,enumerable:!0,configurable:!0})}var n8e={};r8e(n8e,"serializeRPCMessage",()=>ON);r8e(n8e,"deserializeRPCMessage",()=>IN);var PN="PrismaBigInt::",AN="PrismaBytes::";function ON(e){return JSON.stringify(e,(r,n)=>typeof n=="bigint"?PN+n:n?.type==="Buffer"&&Array.isArray(n?.data)?AN+t8e.Buffer.from(n.data).toString("base64"):n)}function IN(e){return JSON.parse(e,(r,n)=>typeof n=="string"&&n.startsWith(PN)?BigInt(n.substr(PN.length)):typeof n=="string"&&n.startsWith(AN)?n.substr(AN.length):n)}var mbe=M(p8e()),bb=M(tbe()),hbe=M(require("http")),gbe=M(ibe()),vbe=require("zlib");var ao=require("path");var f$=require("crypto"),pbe=M(c$());function l$(e,r,n,i){Object.defineProperty(e,r,{get:n,set:i,enumerable:!0,configurable:!0})}var lbe=typeof globalThis<"u"?globalThis:typeof self<"u"?self:typeof window<"u"?window:typeof global<"u"?global:{},u$={},yb={},Bu=lbe.parcelRequire1308;Bu==null&&(Bu=function(e){if(e in u$)return u$[e].exports;if(e in yb){var r=yb[e];delete yb[e];var n={id:e,exports:{}};return u$[e]=n,r.call(n.exports,n,n.exports),n.exports}var i=new Error("Cannot find module '"+e+"'");throw i.code="MODULE_NOT_FOUND",i},Bu.register=function(r,n){yb[r]=n},lbe.parcelRequire1308=Bu);Bu.register("9lTzd",function(module,exports){l$(module.exports,"guessEnginePaths",()=>guessEnginePaths),l$(module.exports,"guessPrismaClientPath",()=>guessPrismaClientPath);var $5COlq=Bu("5COlq");async function guessEnginePaths({forceBinary,forceLibrary,resolveOverrides}){let queryEngineName,queryEngineType;if(forceLibrary?(queryEngineName=await $5COlq.prismaEngineName("query-engine","library"),queryEngineType="library"):forceBinary?(queryEngineName=await $5COlq.prismaEngineName("query-engine","binary"),queryEngineType="binary"):(queryEngineName=void 0,queryEngineType=void 0),!queryEngineName||!queryEngineType)return{queryEngine:void 0};let queryEnginePath;if(resolveOverrides[".prisma/client"])queryEnginePath=(0,ao.resolve)(resolveOverrides[".prisma/client"],`../${queryEngineName}`);else if(resolveOverrides["@prisma/engines"])queryEnginePath=(0,ao.resolve)(resolveOverrides["@prisma/engines"],`../../${queryEngineName}`);else{let atPrismaEnginesPath;try{atPrismaEnginesPath=eval("require.resolve('@prisma/engines')")}catch(e){throw new Error("Unable to resolve Prisma engine paths. This is a bug.")}queryEnginePath=(0,ao.resolve)(atPrismaEnginesPath`../../${queryEngineName}`)}return{queryEngine:{type:queryEngineType,path:queryEnginePath}}}function guessPrismaClientPath({resolveOverrides}){let prismaClientPath=resolveOverrides["@prisma/client"]||eval("require.resolve('@prisma/client')");return(0,ao.resolve)(prismaClientPath,"../")}});Bu.register("5COlq",function(e,r){l$(e.exports,"prismaEngineName",()=>n);async function n(i,a){let o=await Ir(),c=o==="windows"?".exe":"";if(a==="library")return us(o,"fs");if(a==="binary")return`${i}-${o}${c}`;throw new Error(`Unknown engine type: ${a}`)}});function Xyt(e){return{models:p$(e.models),enums:p$(e.enums),types:p$(e.types)}}function p$(e){let r={};for(let{name:n,...i}of e)r[n]=i;return r}var Wg=(0,pbe.debug)("prisma:studio-pcw"),Jyt=/^\s*datasource\s+([^\s]+)\s*{/m,Qyt=/url *= *env\("(.*)"\)/,Zyt=/url *= *"(.*)"/;async function e5t({schema:e,schemaPath:r,dmmf:n,datasourceProvider:i,previewFeatures:a,datasources:o,engineType:c,paths:u,directUrl:p,versions:l}){let f=e.match(Jyt)?.[1]??"",g=e.match(Qyt)?.[1]??null,v=e.match(Zyt)?.[1]??null,{getPrismaClient:E,PrismaClientKnownRequestError:b,PrismaClientRustPanicError:S,PrismaClientInitializationError:D,PrismaClientValidationError:C}=require(`${u.prismaClient}/runtime/${c}`),O=e,I=(0,f$.createHash)("sha256").update(
Raw query failed. Code: `unknown`. Message: `Kind: Command failed: Error code 8000 (AtlasError): empty database name not allowed
	at 774bae2bc957594bbd34308c7c1c95e5/atlasproxy/session_proxy.go:732
	at 774bae2bc957594bbd34308c7c1c95e5/atlasproxy/session_proxy.go:1638
	at 774bae2bc957594bbd34308c7c1c95e5/atlasproxy/session_proxy.go:1213
	at 10gen/mongonet@v0.0.0-20240307160411-5a60262cff18/proxy_session.go:369
	at 10gen/mongonet@v0.0.0-20240307160411-5a60262cff18/proxy_session.go:118
	at 10gen/mongonet@v0.0.0-20240307160411-5a60262cff18/session.go:220
	at 10gen/mongonet@v0.0.0-20240307160411-5a60262cff18/server.go:384
	at src/runtime/asm_arm64.s:1197, labels: {}`
  
Query:
{
  "modelName": "User",
  "operation": "findMany",
  "args": {
    "take": 100,
    "skip": 0,
    "select": {
      "id": true,
      "name": true,
      "email": true,
      "password": true,
      "createdAt": true,
      "updatedAt": true
    }
  }
}
  
```
### How to reproduce

Quando eu rodo o comando: npx prisma studio, me acusa o erro sendo que, ao executar anteriormente o comando npx prisma generate ele fala que tudo ocorreu bem, então não era pra isso ter acontecido. Conforme abaixo:

npx prisma generate

Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
✔ Generated Prisma Client (v5.12.1) to .\node_modules\@prisma\client in 127ms

Start using Prisma Client in Node.js (See: https://pris.ly/d/client)
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)
```
import { PrismaClient } from '@prisma/client/edge'
const prisma = new PrismaClient()
```

See other ways of importing Prisma Client: http://pris.ly/d/importing-client

┌────────────────────────────────────────────────────────────────┐
│  Supercharge your Prisma Client with global database caching,  │
│  scalable connection pooling and real-time database events.    │
│  Explore Prisma Accelerate: https://pris.ly/cli/-accelerate    │
│  Explore Prisma Pulse: https://pris.ly/cli/-pulse 

### Expected behavior

O comportamento esperado era que houvesse conexão com banco de dados MongoDb, onde apesar de falar que quando foi executado npx prisma generate ocorreu tudo bem quando eu acesso o bd não vejo nenhum esquema criado, fora que o studio prisma me da o erro. eu não consigo fazer minhas solicitações POST.

A string de conexão esta correta porque testei usando MongoDB Compass

### Prisma information

// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

// Looking for ways to speed up your queries, or scale easily with your serverless or edge functions?
// Try Prisma Accelerate: https://pris.ly/cli/accelerate-init
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  name      String
  email     String   @unique
  password  String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@map("users")
}

model Day {
  id     String   @id @default(auto()) @map("_id") @db.ObjectId
  date   DateTime @unique
  tables Table[]
}

model Reservation {
  id      String  @id @default(auto()) @map("_id") @db.ObjectId
  name    String
  phone   String?
  email   String?
  table   Table?
  tableId String?
}

model Table {
  id            String       @id @default(auto()) @map("_id") @db.ObjectId
  name          String
  capacity      Int
  isAvailable   Boolean      @default(true)
  location      String
  reservationId String       @unique @map("reservation_id") @db.ObjectId
  reservation   Reservation? @relation(fields: [reservationId], references: [id])
  dayId         String       @map("day_id") @db.ObjectId
  day           Day          @relation(fields: [dayId], references: [id])
}

```
### Environment & setup

- OS: Windows 10
- Database: MongoDB
- Node.js version: 18.17.1

### Prisma Version

prisma: 5.12.1

**Comment by ludralph** [comment]
Hi @nathanmota-dev 👋 

Thank you for raising this question.

The error message indicates that the database name in your connection string is empty. Prisma Client requires a database name to be specified in the connection string to connect and interact with the database. 

```
mongodb+srv://user:password@cluster0.cavf0vi.mongodb.net/DATABASE_NAME?retryWrites=true&w=majority
```

Replace `DATABASE_NAME` with the name of your database. If the database name is not provided, you might encounter errors as Prisma Client won't know which database to connect to.

**Comment by nathanmota-dev** [comment]
The error I am experiencing is not with the connection string; the connection string is correct because I tested it using Atlas Compass. Additionally, I conducted another backend test using Mongoose instead of Prisma and did not encounter any problems, so the error is not with the connection string but rather with Prisma. It seems as though the command 'npx prisma generate' is not working. I have also tried uninstalling and reinstalling Prisma and prisma/client dependencies, but it didn't help; the error persists.

**Reply by ludralph** [reply]
Hi @nathanmota-dev 

From the error message:
> Raw query failed. Code: `unknown`. Message: `Kind: Command failed: Error code 8000 (AtlasError): empty database name not allowed

You need to append the database name to the connection string for MongoDB Atlas. See this [documentation](https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/mongodb/connect-your-database-typescript-mongodb#raw-query-failed-error-code-8000-atlaserror-empty-database-name-not-allowed).

**Comment by dbracken-735** [comment]
I think I've seen this bug before. Before the error occurred, did you make a change to the DB schema with prisma studio still running?

**Reply by DubG1** [reply]
this happened to me i just implemented an isnert function for my db and i am using prisma studio to view my data and delete records, but now i changed some of the tables and if i want to delete a table i get a pretty long error message similiar to here, i am trying now to delete indexedDB now like in the answer to @nathanmota-dev

**Reply by DubG1** [reply]
i tried a lot and looked up some stuff but couldnt fix it yet

**Comment by nathanmota-dev** [comment]
I think I did the same thing, do you know how to fix it? I even uninstalled the dependencies and installed them again but it didn't solve the problem.

**Reply by dbracken-735** [reply]
I'm not at my computer to access the whole workflow but here are the concepts I can remember; sorry to the length. 

If you have prisma studio running in vs code on windows 10/11 and run prisma generate, prisma studio sometimes throws a fatal, irreconcilable error. The issue isn't with your browser, your npm's or node -- it's windows. 

if you google deleting indexedDB google chrome, and follow the steps you find to delete the indexedDB file(s) cached by windows, it should clear the error. 

Clearing your web history / cache through the browser won't do it -- you have to clear it using the windows file system. Deleting WSL, node, or similar things won't do it either -- only deleting it from the windows file system. 

Please let me know if this works for you. If not, I can get you the exact workflow in the morning.

**Comment by yuri-kerven** [comment]
ola, estou aprendendo ainda, mais consegui acha o erro do meu codigo, que no caso estava dando error pois, mudei o parametro de "Password" para "password",

![image](https://github.com/user-attachments/assets/14aed853-10d5-4a1a-8e57-1efee226f2f6)


 com isso o prisma, não conseguia identificar e gerava os erros, fui no Mongodb e apaguei os dados

![image](https://github.com/user-attachments/assets/be5e5102-e33e-4e20-ac4c-b46792d9dfb9)

e então voltou, sem erro. por algum motivo o npx prisma generate ou o push não funciona....

**Comment by ankhyk** [comment]
I'm using sqlite and this error happened to me, and it have been fixed by using a absolute path in the `DATABASE_URL` in file .env instead of a relative path.

If prisma studio cannot find the db file , it will create a new db file in a wrong path and read it, so it cannot find the database name.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
